### ISSUE TITLE
Unable to resume activity in release mode Java.lang.NoSuchMethodException

### ISSUE THREAD

**Issue by OmarACY** [mono0926/barcode_scan2#36]
**Describe the bug**
I published an app in play store and throw an java.lang.NoSuchMethodException. I attached the trace in **Logs** section
The issue also happens when you run the app with the follow command:
`flutter run --release`
I took the example of library and the issue is the same.

**To Reproduce**
The method BarcodeScanner.scan(); throws an exception in production.
```
  void _scanQR(BuildContext context) async {
    ScanResult? scanResult;

    try {
      scanResult = await BarcodeScanner.scan();
    } on PlatformException catch (e) {
      if (e.code == BarcodeScanner.cameraAccessDenied) {
        Fluttertoast.showToast(msg: 'The camera access was denied');
      } else {
        ...
      }
    } catch (e) {
      ...
    }

    if (scanResult != null && scanResult.rawContent != '') {
     ...
    }
  }
```

**Expected behavior**
The call to scan should return and valid ScanResult

**Logs**
```
java.lang.RuntimeException: Unable to resume activity {###.####.###/de.mintware.barcode_scan.BarcodeScannerActivity}: java.lang.RuntimeException: java.lang.NoSuchMethodException: c.b.d.e.values []
java.lang.RuntimeException: 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4504)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4536)
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:52)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2104)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:236)
  at android.app.ActivityThread.main (ActivityThread.java:7861)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:600)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:967)
Caused by: java.lang.RuntimeException: 
  at java.lang.Enum.enumValues (Enum.java:270)
  at java.lang.Enum.access$000 (Enum.java:61)
  at java.lang.Enum$1.create (Enum.java:277)
  at java.lang.Enum$1.create (Enum.java:275)
  at libcore.util.BasicLruCache.get (BasicLruCache.java:63)
  at java.lang.Enum.getSharedConstants (Enum.java:289)
  at java.lang.Class.getEnumConstantsShared (Class.java:2428)
  at java.util.EnumMap.getKeyUniverse (EnumMap.java:755)
  at java.util.EnumMap.<init> (EnumMap.java:138)
  at me.dm7.barcodescanner.zxing.ZXingScannerView.o (ZXingScannerView.java:4)
  at me.dm7.barcodescanner.zxing.ZXingScannerView.<init> (ZXingScannerView.java:3)
  at de.mintware.barcode_scan.ZXingAutofocusScannerView.<init> (ZXingAutofocusScannerView.java:5)
  at de.mintware.barcode_scan.BarcodeScannerActivity.setupScannerView (BarcodeScannerActivity.java)
  at de.mintware.barcode_scan.BarcodeScannerActivity.onResume (BarcodeScannerActivity.java)
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1457)
  at android.app.Activity.performResume (Activity.java:8264)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4494)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4536)
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:52)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2104)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:236)
  at android.app.ActivityThread.main (ActivityThread.java:7861)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:600)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:967)
Caused by: java.lang.NoSuchMethodException: 
  at java.lang.Class.getMethod (Class.java:2072)
  at java.lang.Class.getDeclaredMethod (Class.java:2050)
  at java.lang.Enum.enumValues (Enum.java:267)
  at java.lang.Enum.access$000 (Enum.java:61)
  at java.lang.Enum$1.create (Enum.java:277)
  at java.lang.Enum$1.create (Enum.java:275)
  at libcore.util.BasicLruCache.get (BasicLruCache.java:63)
  at java.lang.Enum.getSharedConstants (Enum.java:289)
  at java.lang.Class.getEnumConstantsShared (Class.java:2428)
  at java.util.EnumMap.getKeyUniverse (EnumMap.java:755)
  at java.util.EnumMap.<init> (EnumMap.java:138)
  at me.dm7.barcodescanner.zxing.ZXingScannerView.o (ZXingScannerView.java:4)
  at me.dm7.barcodescanner.zxing.ZXingScannerView.<init> (ZXingScannerView.java:3)
  at de.mintware.barcode_scan.ZXingAutofocusScannerView.<init> (ZXingAutofocusScannerView.java:5)
  at de.mintware.barcode_scan.BarcodeScannerActivity.setupScannerView (BarcodeScannerActivity.java)
  at de.mintware.barcode_scan.BarcodeScannerActivity.onResume (BarcodeScannerActivity.java)
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1457)
  at android.app.Activity.performResume (Activity.java:8264)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4494)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4536)
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:52)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2104)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:236)
  at android.app.ActivityThread.main (ActivityThread.java:7861)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:600)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:967)
```

**Environment (please complete the following information):**
 - [ ] iOS? Version: ...
 - [ ] Android? Version:   Android 11 (SDK 30)
 - [ ] Real Device? Which?: Redmi Note 9 Pro
 - [ ] Emulator/Simulator? Which?: ...
 - Version of `barcode_scan`:  ^4.2.0
 - Output of `flutter --version`:
    Flutter 2.5.2 • channel stable • https://github.com/flutter/flutter.git
    Framework • revision 3595343e20 (8 weeks ago) • 2021-09-30 12:58:18 -0700
    Engine • revision 6ac856380f
    Tools • Dart 2.14.3

**Additional context**
I've worked with this library in test and it works, i found the problem in the production app in the [Play Store](https://play.google.com/store/apps/details?id=com.omaracy.qrtools).

**Comment by OmarACY** [comment]
I tried solve it again but i don't know what is the reason of the problem, i updated the error message it is the same but with more information at the beginning.

**Comment by OmarACY** [comment]
I took the example of library and the issue is the same.

**Comment by OmarACY** [comment]
Finally i fixed it.

1. I added this file `android/app/proguard-rules.pro`  and i added the code as follow:
```
-dontobfuscate
-dontshrink
```
2. In the `project/app/build.gradle` file, I added the code as follow:
```
android {
   ...

    buildTypes {
        release {
            ...
            // add new line
            minifyEnabled true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'

        }
    }
   ...
}
```

3. I executed the command `flutter run --release` and it works! also works with `flutter build apk --release`

This question helped me 
[flutter-build-release-apk-get-nosuchmethodexception-error](https://stackoverflow.com/questions/61699344/flutter-build-release-apk-get-nosuchmethodexception-error/61708331#61708331)

And it was not an error of the library
